### ISSUE TITLE
add negate flag to Minesweeper

### ISSUE THREAD

**Issue by anothermattbrown** [batfish/batfish#831]
Running `smt-reachability` with `finalNodeRegex` can return flows where the final node does not match the parameter value. 

```
> init-testrig test_rigs/smt-examples/BGP1 tr-smt-bgp1
> get smt-reachability finalNodeRegex="R2", dstIps=['42.42.42.1']
...
Question: smt-reachability actions=[ACCEPT], dstIps=[42.42.42.1], finalNodeRegex=R2

Flow: ingress:R1 vrf:default 0.0.0.0->42.42.42.1 HOPOPT packetLength:0 state:NEW
  environment:BASE
Environment{envName=BASE, testrigName=tr-smt-bgp1, edgeBlacklist=null, interfaceBlacklist=null, nodeBlacklist=null, bgpTables=null, routingTables=null, externalBgpAnnouncements=null}
    ACCEPTED

Flow: ingress:R2 vrf:default 0.0.0.0->42.42.42.1 HOPOPT packetLength:0 state:NEW
  environment:BASE
Environment{envName=BASE, testrigName=tr-smt-bgp1, edgeBlacklist=null, interfaceBlacklist=null, nodeBlacklist=null, bgpTables=null, routingTables=null, externalBgpAnnouncements=null}
    Hop 1: R2:Serial0 -> R1:Serial0 --- [BgpRoute<42.42.42.0/24,nhip:192.168.42.2,nhint:dynamic>]
    ACCEPTED
```

Here, traffic in the each flow terminates at R1, which does not match the `finalNodeRegex` R2.

A second problem (which might need to be field as a separate bug) is that omitting the `finalNodeRegex` parameter causes zero flows to be reported, when the two above flows should be:

```
> get smt-reachability dstIps=['42.42.42.1']
Question: smt-reachability actions=[ACCEPT], dstIps=[42.42.42.1]

Verified
```

`finalNodeRegex` is used in a `smt-reachability` test in the `java-smt/commands` test suite, which would indicate that it is supported.

**Comment by anothermattbrown** [comment]
According @rabeckett, the question: `smt-reachability actions=[ACCEPT], dstIps=[42.42.42.1], finalNodeRegex=R2` means "all accepted traffic for the IP will have the finalNode matching the regex". Minesweeper gives us counterexamples of that assertion.

**Comment by anothermattbrown** [comment]
In order to get results comparable to the `reachability` query, we can add a boolean `negate` parameter to the query, which would change the meaning of the query from "verify this assertion or show me counterexamples" to "show me examples of flows with this property". In other words, it would make `smt-reachability` verify *un*reachability properties. 

For example, on the `smt-examples/BGP1` network, if we run `get smt-reachability dstIps=['42.42.42.1']`, the result is "Verified", indicating that all nodes can reach that IP. If we run `get smt-reachability dstIps=['42.42.42.1'], negate=true`, we are asking Minesweeper to verify that no node can reach the IP. This is false, and we should get one counterexample for each node. We should then be able to validate each counterexample using the `reachability` query.

**Comment by dhalperi** [comment]
@anothermattbrown is this done?

**Comment by anothermattbrown** [comment]
Yes, fixed by PR  #852.
